src/kps/Main.java

```diff
@@ -23,6 +24,9 @@ public static void main(String[] args) throws XMLException, FileNotFoundExceptio
         Simulation simulation;
         try {
             simulation = SimulationXML.readSimulationFromFile(new FileInputStream(args[0]));
+            for (Mail m : simulation.getMail()){
+                m.setCalculatedRoute(simulation.buildCalculatedRoute(m.getFrom(), m.getTo(), m.getPriority()));
+            }
```

Comment:
I don't understand why you need to calculate routes on startup. They should be stored in the xml